src/Speckle.Sdk/Serialisation/V2/SerializeProcessFactory.cs

```diff
@@ -46,34 +27,54 @@ public ISerializeProcess CreateSerializeProcess(
   {
     var sqLiteJsonCacheManager = sqLiteJsonCacheManagerFactory.CreateFromStream(streamId);
     var serverObjectManager = serverObjectManagerFactory.Create(url, streamId, authorizationToken);
-    return new SerializeProcess(
+    return CreateSerializeProcess(sqLiteJsonCacheManager, serverObjectManager, progress, cancellationToken, options);
+  }
+
+  public ISerializeProcess CreateSerializeProcess(
+    ISqLiteJsonCacheManager sqLiteJsonCacheManager,
+    IServerObjectManager serverObjectManager,
+    IProgress<ProgressArgs>? progress,
+    CancellationToken cancellationToken,
+    SerializeProcessOptions? options = null
+  ) =>
+    new SerializeProcess(
       progress,
-      sqLiteJsonCacheManager,
-      serverObjectManager,
+      new ObjectSaver(
+        progress,
+        sqLiteJsonCacheManager,
+        serverObjectManager,
+        loggerFactory.CreateLogger<ObjectSaver>(),
+        cancellationToken
+      ),
       baseChildFinder,
       new BaseSerializer(sqLiteJsonCacheManager, objectSerializerFactory),
       loggerFactory,
       cancellationToken,
       options
     );
-  }
 
-  public IDeserializeProcess CreateDeserializeProcess(
-    Uri url,
-    string streamId,
-    string? authorizationToken,
+  public ISerializeProcess CreateSerializeProcess(
+    ConcurrentDictionary<Id, Json> jsonCache,
+    ConcurrentDictionary<string, string> objects,
     IProgress<ProgressArgs>? progress,
     CancellationToken cancellationToken,
-    DeserializeProcessOptions? options = null
+    SerializeProcessOptions? options = null
   )
   {
-    var sqLiteJsonCacheManager = sqLiteJsonCacheManagerFactory.CreateFromStream(streamId);
-    var serverObjectManager = serverObjectManagerFactory.Create(url, streamId, authorizationToken);
-    return new DeserializeProcess(
-      sqLiteJsonCacheManager,
-      serverObjectManager,
+#pragma warning disable CA2000
+    var memoryJsonCacheManager = new MemoryJsonCacheManager(jsonCache);
```

Comment:
Why not use a `using` statement? instead of supressing?

Reply:
can't put something that will be owned by the factory result in a using statement.  The result object owns it.

Reply:
Supprised this gives a CA2000. normally that permits transfer of owership via constructor...
CA2000 is notoriously unreliable, so happy with current state.